cardano-api/src/Cardano/Api/KeysByron.hs

```diff
@@ -54,6 +68,127 @@ import           Cardano.Api.SerialiseTextEnvelope
 -- This is a type level tag, used with other interfaces like 'Key'.
 --
 data ByronKey
+data ByronKeyLegacy
+
+class IsByronLegacyFormat key where
+  isLegacyFormat :: ByronKeyFormat key
+
+data ByronKeyFormat key where
+  ByronLegacyKeyFormat :: ByronKeyFormat ByronKeyLegacy
+  ByronNonLegacyKeyFormat :: ByronKeyFormat ByronKey
+
+instance IsByronLegacyFormat ByronKey where
+  isLegacyFormat = ByronNonLegacyKeyFormat
+
+instance IsByronLegacyFormat ByronKeyLegacy where
+  isLegacyFormat = ByronLegacyKeyFormat
+
+instance HasTypeProxy ByronKeyLegacy where
```

Comment:
We should probably do all the ByronKeyLegacy instances after ByronKey in this module, just for ease of reading. So yes the two tag data types (ByronKey and ByronKeyLegacy) first, then the classes and types that cover both (class IsByronKey and data ByronKeyFormat), then sections for the impl for the specific key types, ByronKey then ByronKeyLegacy. It's ok to use section separators like we do in other modules.